packages/regl-worldview/src/WorldviewContext.js

```diff
@@ -239,61 +242,110 @@ export class WorldviewContext {
+            // https://github.com/regl-project/regl/blob/28fbf71c871498c608d9ec741d47e34d44af0eb5/lib/read.js#L57
+            if (x < Math.floor(width) && y < Math.floor(height) && x >= 0 && y >= 0) {
+              const pixel = new Uint8Array(4);
+
+              // read pixel value from the frame buffer
+              regl.read({
+                x,
+                y,
+                width: 1,
+                height: 1,
+                data: pixel,
+              });
+
+              currentObjectId = getIdFromPixel(pixel);
+              const mouseEventObject = this._hitmapObjectIdManager.getObjectByObjectHitmapId(currentObjectId);
+              // Check an error case: if we've already seen this hitmapObjectId, then the getHitmapFromChildren function
+              // is not respecting the excludedObjects correctly and we should notify the user of a bug.
+              if (seenObjectIds.has(currentObjectId)) {
+                const command =
+                  mouseEventObject.object != null
+                    ? this._hitmapObjectIdManager.getCommandForObject(mouseEventObject.object)
+                    : null;
+                const displayName = command != null ? command.displayName : "UNKNOWN_COMMAND";
+                console.error(
+                  `Saw object twice when reading from hitmap. There is likely an error in getHitmapFromChildren for ${displayName}.`,
+                  mouseEventObject
+                );
+                break;
+              }
+              seenObjectIds.add(currentObjectId);
+
+              if (currentObjectId > 0 && mouseEventObject.object) {
+                const command = this._hitmapObjectIdManager.getCommandForObject(mouseEventObject.object);
+                excludedObjects.push(mouseEventObject);
+                if (command) {
+                  mouseEventsWithCommands.push([mouseEventObject, command]);
+                }
+              }
+            }
+            // If we haven't enabled stacked object events, break out of the loop immediately.
+            // eslint-disable-next-line no-unmodified-loop-condition
+          } while (currentObjectId !== 0 && enableStackedObjectEvents);
+
+          resolve(mouseEventsWithCommands);
         });
       });
     });
   }
 
-  callComponentHandlers = (objectId: number, ray: Ray, e: MouseEvent, mouseEventName: MouseEventEnum) => {
-    this._hitmapCalls.forEach((_, component) => {
-      if (component instanceof Command) {
-        component.handleMouseEvent(objectId, e, ray, mouseEventName);
-      }
-    });
-  };
-
-  _drawInput = (isHitmap?: boolean) => {
-    const drawCallsMap = isHitmap ? this._hitmapCalls : this._drawCalls;
-    const sortedDrawCalls = Array.from(drawCallsMap.values()).sort((a, b) => (a.layerIndex || 0) - (b.layerIndex || 0));
+  _drawInput = (isHitmap?: boolean, excludedObjects?: MouseEventObject[]) => {
+    if (isHitmap) {
+      this._hitmapObjectIdManager.reset();
```

Comment:
Maybe just create a new manager object each time? otherwise there's some risk of bugs if we add some behavior and forget to clear it in reset().